es-design-system/pages/examples/form-validation.vue

```diff
@@ -0,0 +1,124 @@
+                    </b-link> for field level validation example
+                </p>
+                <h2>
+                    UX Guidelines
+                </h2>
+                <p>
+                    <ul>
+                        <li>
+                            Show loading spinner on button during server request
+                        </li>
+                        <li>
+                            Use <b-link :to="{ name: 'molecules-es-form-msg___en' }">
+                                EsFormMsg
+                            </b-link>
+                            for displaying of server-side form-level errors
+                        </li>
+                    </ul>
+                </p>
+            </b-col>
+        </b-row>
+        <b-row class="border-top pt-4 my-2">
+            <b-col
+                cols="12"
+                lg="8">
+                <b-form
+                    @submit.stop.prevent="onSubmit">
+                    <es-form-msg
+                        class="mt-3"
+                        :variant="formMsgVariant"
+                        :message="formMsg"
+                        @hidden="formMsg = ''" />
+                    <es-form-input
+                        id="form-input-name"
+                        v-model="form.name"
+                        :disabled="isSubmitInProgress"
+                        required>
+                        <template #label>
+                            Name
+                        </template>
+                    </es-form-input>
+                    <div class="d-flex flex-grow-1 justify-content-end mt-3">
+                        <es-button
+                            type="submit"
+                            class="w-100 w-lg-auto"
+                            :loading="isSubmitInProgress"
+                            :disabled="isSubmitInProgress">
+                            <span class="w-100 min-width-6">
+                                <span
+                                    v-if="isSubmitInProgress">
+                                    <b-spinner
+                                        role="status"
+                                        aria-hidden="true"
+                                        label="Loading" />
+                                </span>
+                                <span
+                                    :class="{'sr-only': isSubmitInProgress }">
+                                    Submit
+                                </span>
+                            </span>
+                        </es-button>
+                    </div>
+                </b-form>
+            </b-col>
+        </b-row>
+    </b-container>
+</template>
+
+<script>
+import {
+    EsFormInput,
+    EsButton,
+    EsFormMsg,
+    formMixins,
+} from '@energysage/es-vue-base';
+
+export default {
+    name: 'EsFormDocs',
+    components: {
+        EsFormMsg,
+        EsFormInput,
+        EsButton,
+    },
+    mixins: [formMixins],
```

Comment:
With the additional error checking in the `formMixins` the method `startSubmit`, `stopSubmit`, and `showFormError` can be called for form-level validation, without needing to included any client-level validation location i.e. adding the `validators` section required for vuelidate.